src/txrebroadcast.cpp

```diff
@@ -7,17 +7,22 @@
 #include <miner.h>
 #include <script/script.h>
 #include <txrebroadcast.h>
+#include <util/time.h>
 
 /** We rebroadcast 3/4 of max block weight to reduce noise due to circumstances
  *  such as miners mining priority transactions. */
 static constexpr unsigned int MAX_REBROADCAST_WEIGHT = 3 * MAX_BLOCK_WEIGHT / 4;
 
+/** Default minimum age for a transaction to be rebroadcast */
+static constexpr std::chrono::minutes REBROADCAST_MIN_TX_AGE = 30min;
```

Comment:
Any reasons for 30 minutes? Less than 30 minutes might have issues. What if we use more than 30 minutes or user can define this in `bitcoin.conf`?

Reply:
30 minutes was chosen as the minimum age to ensure that transactions would have ample time to propagate through mempools. I don't think it makes sense for this to be a configurable value, both because of what sipa highlighted here: https://github.com/bitcoin/bitcoin/pull/21061#issuecomment-806284509, as well as the fact that it would increase the complexity of reasoning about & maintenance burden. However, that said, I don't think of this value (30min) as set in stone- through the review process & running the patch, we should identify if this value is appropriate. Another thing to keep in mind is that any rebroadcast candidates will have to pass through `filterInventoryKnown` for an inv to be sent to the peer.